rust/lance-index/src/vector/pq.rs

```diff
@@ -57,7 +57,7 @@ pub trait ProductQuantizer: Send + Sync + std::fmt::Debug {
     async fn transform(&self, data: &dyn Array) -> Result<ArrayRef>;
 
     /// Build the distance lookup in `f32`.
-    fn build_distance_table(&self, query: &dyn Array, code: &UInt8Array) -> Result<ArrayRef>;
+    fn build_distance_table(&self, query: &dyn Array, code: &UInt8Array) -> Result<Float32Array>;
```

Comment:
This will always be f32? Or does it need to be generified ?

Reply:
The return from distance functions are all `f32` already. 

https://github.com/lancedb/lance/blob/27cf463365eaee35c94b087d6ad8884dc5e4068d/rust/lance-index/src/vector/pq.rs#L286